### PR TITLE
[tests] Fix creation of IntPtr to work on a 32-bit system.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

File: tests/monotouch-test/ObjCRuntime/RegistrarTest.cs [hunk 1]
```diff
@@ -1286,7 +1286,7 @@ public void TestConstrainedGenericType ()
 						void_objc_msgSend_out_IntPtr (obj.Handle, Selector.GetHandle ("m2:"), out value);
 						Assert.AreEqual (IntPtr.Zero, value);
 
-						value = new IntPtr (0xdeadbeef);
+						value = new IntPtr ((unchecked ((int) 0xdeadbeef)));
```
Comment (spouliot) [inline tests/monotouch-test/ObjCRuntime/RegistrarTest.cs:1289]: does it make it an `un` `deadbeef` ?

**Review by spouliot — APPROVED** [review]

File: tests/monotouch-test/ObjCRuntime/RegistrarTest.cs (hunk 1 above)
Comment (rolfbjarne) [reply tests/monotouch-test/ObjCRuntime/RegistrarTest.cs:1289]: Only 32% dead as opposed to 64% dead now.

Comment (spouliot) [reply tests/monotouch-test/ObjCRuntime/RegistrarTest.cs:1289]: moar 🧟

**Review by dalexsoto — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failures are unrelated:

* xcframework-test/Mac Catalyst [dotnet]/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2414
* link sdk/tvOS - simulator/Debug: https://github.com/xamarin/xamarin-macios/issues/12640
* link sdk/tvOS - simulator/Release: https://github.com/xamarin/xamarin-macios/issues/12640

**Pull request merged**